<launch>
	<arg name="sub_msg_to_robot_topic_name"       default="/handyman/message/to_robot" />
	<arg name="pub_msg_to_moderator_topic_name"   default="/handyman/message/to_moderator" />
	<arg name="sub_joint_state_topic_name"        default="/hsrb/joint_states" />
	<arg name="pub_base_twist_topic_name"         default="/hsrb/command_velocity" />
	<arg name="pub_arm_trajectory_topic_name"     default="/hsrb/arm_trajectory_controller/command" />
	<arg name="pub_gripper_trajectory_topic_name" default="/hsrb/gripper_controller/command" />
	
	<arg name="sub_laser_scan_topic_name"         default="/hsrb/base_scan" />
	<arg name="rgbd_camera"                       default="head_rgbd_sensor" />
	
	<arg name="sigverse_ros_bridge_port"        default="50001" />
	<arg name="ros_bridge_port"                 default="9090" />

	<node name="teleop_key_handyman" pkg="handyman" type="teleop_key_handyman" output="screen" launch-prefix="xterm -e">

		<param name="sub_msg_to_robot_topic_name"       type="string" value="$(arg sub_msg_to_robot_topic_name)" />
		<param name="pub_msg_to_moderator_topic_name"   type="string" value="$(arg pub_msg_to_moderator_topic_name)" />
		<param name="sub_joint_state_topic_name"        type="string" value="$(arg sub_joint_state_topic_name)" />
		<param name="pub_base_twist_topic_name"         type="string" value="$(arg pub_base_twist_topic_name)" />
		<param name="pub_arm_trajectory_topic_name"     type="string" value="$(arg pub_arm_trajectory_topic_name)" />
		<param name="pub_gripper_trajectory_topic_name" type="string" value="$(arg pub_gripper_trajectory_topic_name)" />
	</node>

	<group ns="sigverse_ros_bridge">
		<node name="sigverse_ros_bridge" pkg="sigverse_ros_bridge" type="sigverse_ros_bridge" args="$(arg sigverse_ros_bridge_port)"/>
	</group>
	
	<include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" > 
		<arg name="port" value="$(arg ros_bridge_port)"/>
	</include>
	
	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find handyman)/launch/hsr.rviz"/>
	

	<!-- Gmapping -->
<!--
	<node pkg="gmapping" type="slam_gmapping" name="sigverse_hsr_slam">
		<remap from="scan" to="$(arg sub_laser_scan_topic_name)"/>
		
		<param name="base_frame" value="base_footprint"/>
		<param name="odom_frame" value="odom"/>
		<param name="map_update_interval" value="0.1"/>
		<param name="maxUrange" value="4.0"/>
		<param name="minimumScore" value="100"/>
		<param name="linearUpdate" value="0.2"/>
		<param name="angularUpdate" value="0.2"/>
		<param name="temporalUpdate" value="0.5"/>
		<param name="delta" value="0.05"/>
		<param name="lskip" value="0"/>
		<param name="particles" value="30"/>
		<param name="sigma" value="0.05"/>
		<param name="kernelSize" value="1"/>
		<param name="lstep" value="0.05"/>
		<param name="astep" value="0.05"/>
		<param name="iterations" value="5"/>
		<param name="lsigma" value="0.075"/>
		<param name="ogain" value="3.0"/>
		<param name="srr" value="0.01"/>
		<param name="srt" value="0.02"/>
		<param name="str" value="0.01"/>
		<param name="stt" value="0.02"/>
		<param name="resampleThreshold" value="0.5"/>
		<param name="xmin" value="-10.0"/>
		<param name="ymin" value="-10.0"/>
		<param name="xmax" value="10.0"/>
		<param name="ymax" value="10.0"/>
		<param name="llsamplerange" value="0.01"/>
		<param name="llsamplestep" value="0.01"/>
		<param name="lasamplerange" value="0.005"/>
		<param name="lasamplestep" value="0.005"/>
	</node>
	
	<group ns="hsrb">
		<include file="$(find realsense_camera)/launch/r200_nodelet_rgbd.launch">
			<arg name="camera"      value="$(arg rgbd_camera)"/>
			<arg name="load_driver" value="false"/>
		</include>
	</group>
-->
	
</launch>

